src/ScopedValues.jl

```diff
@@ -165,6 +202,14 @@ function with(f, pair::Pair{<:ScopedValue}, rest::Pair{<:ScopedValue}...)
     end
 end
 
+function with(f, snapshot::Snapshot)
+    @nospecialize
+    scope = restore(snapshot)
+    enter_scope(scope) do
+        f()
+    end
```

Comment:
This do- lambda is quite probably free, or nearly free, but maybe this would be simpler to pass f in to enter_scope?
```suggestion
    enter_scope(f, scope)
```
I think your current version reads nicer though, so i dunno what's better.

Reply:
I think both are fine, but we should probably use `enter_scope(f, scope)`.